model/integration_action.go

```diff
@@ -25,15 +25,49 @@ const (
 
 type DoPostActionRequest struct {
 	SelectedOption string `json:"selected_option"`
+	Cookie         string `json:"cookie,omitempty"`
 }
 
 type PostAction struct {
-	Id          string                 `json:"id"`
-	Name        string                 `json:"name"`
-	Type        string                 `json:"type"`
-	DataSource  string                 `json:"data_source"`
-	Options     []*PostActionOptions   `json:"options"`
+	// A unique Action ID. If not set, generated automatically.
+	Id string `json:"id"`
+
+	// The type of the interactive element. Currently supported are
+	// "select" and "button".
+	Type string `json:"type"`
+
+	// The text on the button, or in the select placeholder.
+	Name string `json:"name"`
+
+	// DataSource indicates the data source for the select action. If left
+	// empty, the select is populated from Options. Other supported values
+	// are "users" and "channels".
+	DataSource string               `json:"data_source"`
+	Options    []*PostActionOptions `json:"options"`
+
+	// Defines the interaction with the backend upon a user action.
+	// Integration contains Context, which is private plugin data;
+	// Integrations are stripped from Posts when they are sent to the
+	// client, or are encrypted in a Cookie.
 	Integration *PostActionIntegration `json:"integration,omitempty"`
+	Cookie      string                 `json:"cookie,omitempty" db:"-"`
```

Comment:
What does `db:"-"` do? I've never seen this before.

Reply:
https://github.com/mattermost/mattermost-server/blob/master/model/post.go#L73 for example. Avoids storing this field in the database?